google/resource_sql_database_instance.go

```diff
@@ -640,243 +655,13 @@ func resourceSqlDatabaseInstanceRead(d *schema.ResourceData, meta interface{}) e
-		}
-
-		if len(_locationPreferenceList) == 1 && _locationPreferenceList[0] != nil &&
-			settings.LocationPreference != nil {
-			_locationPreference := _locationPreferenceList[0].(map[string]interface{})
-
-			if vp, okp := _locationPreference["follow_gae_application"]; okp && vp != nil {
-				_locationPreference["follow_gae_application"] =
-					settings.LocationPreference.FollowGaeApplication
-			}
-
-			if vp, okp := _locationPreference["zone"]; okp && vp != nil {
-				_locationPreference["zone"] = settings.LocationPreference.Zone
-			}
-
-			_locationPreferenceList[0] = _locationPreference
-			_settings["location_preference"] = _locationPreferenceList[0]
-		}
-	}
-
-	if v, ok := _settings["maintenance_window"]; ok && len(v.([]interface{})) > 0 &&
-		settings.MaintenanceWindow != nil {
-		_maintenanceWindow := v.([]interface{})[0].(map[string]interface{})
-
-		if vp, okp := _maintenanceWindow["day"]; okp && vp != nil {
-			_maintenanceWindow["day"] = settings.MaintenanceWindow.Day
-		}
-
-		if vp, okp := _maintenanceWindow["hour"]; okp && vp != nil {
-			_maintenanceWindow["hour"] = settings.MaintenanceWindow.Hour
-		}
-
-		if vp, ok := _maintenanceWindow["update_track"]; ok && vp != nil {
-			if len(vp.(string)) > 0 {
-				_maintenanceWindow["update_track"] = settings.MaintenanceWindow.UpdateTrack
-			}
-		}
-	}
-
-	if v, ok := _settings["pricing_plan"]; ok && len(v.(string)) > 0 {
-		_settings["pricing_plan"] = settings.PricingPlan
-	}
-
-	if v, ok := _settings["replication_type"]; ok && len(v.(string)) > 0 {
-		_settings["replication_type"] = settings.ReplicationType
-	}
-
-	_settingsList = make([]interface{}, 1)
-	_settingsList[0] = _settings
-	d.Set("settings", _settingsList)
-
-	if v, ok := d.GetOk("replica_configuration"); ok && v != nil {
-		_replicaConfigurationList := v.([]interface{})
-		if len(_replicaConfigurationList) == 1 && _replicaConfigurationList[0] != nil {
-			_replicaConfiguration := _replicaConfigurationList[0].(map[string]interface{})
-
-			if vp, okp := _replicaConfiguration["failover_target"]; okp && vp != nil {
-				_replicaConfiguration["failover_target"] = instance.ReplicaConfiguration.FailoverTarget
-			}
-
-			// Don't attempt to assign anything from instance.ReplicaConfiguration.MysqlReplicaConfiguration,
-			// since those fields are set on create and then not stored. See description at
-			// https://cloud.google.com/sql/docs/mysql/admin-api/v1beta4/instances
-
-			_replicaConfigurationList[0] = _replicaConfiguration
-			d.Set("replica_configuration", _replicaConfigurationList)
-		}
-	}
-
-	_ipAddresses := make([]interface{}, len(instance.IpAddresses))
-
-	for i, ip := range instance.IpAddresses {
-		_ipAddress := make(map[string]interface{})
-
-		_ipAddress["ip_address"] = ip.IpAddress
-		_ipAddress["time_to_retire"] = ip.TimeToRetire
-
-		_ipAddresses[i] = _ipAddress
-	}
-
-	d.Set("ip_address", _ipAddresses)
+	d.Set("settings", flattenSettings(instance.Settings))
```

Comment:
We generally recommend checking for errors from `d.Set` when setting non scalar values like slices and maps.

```
if err := d.Set("settings", flattenSettings(instance.Settings)); err != nil {
  log.Printf("[WARN] failed to set <...>")
}
```

Generally you want to log a `WARN` and not actually error out here